### PR TITLE
feat(plasma-b2c): add accent color for slider thumb when focus

### PR DESCRIPTION
**Pull request by Yakutoc**
При фокусе меняем **border-color** цвет на **accent**

:baby_chick: Download canary assets:

[colors--canary.205.3375581203.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.205.3375581203.xml)  
[color_sbermarket_ios-swift--canary.205.3375581203.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.205.3375581203.swift)  
[color_sbermarket_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.205.3375581203.kt)  
[color_sbermarket_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.205.3375581203.ts)  
[color_sberprime_ios-swift--canary.205.3375581203.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.205.3375581203.swift)  
[color_sberprime_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.205.3375581203.kt)  
[color_sberprime_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.205.3375581203.ts)  
[PlasmaTokensColor--canary.205.3375581203.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.205.3375581203.swift)  
[typo_mage_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.205.3375581203.kt)  
[typo_mage_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.205.3375581203.ts)  
[typo_plasma_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.205.3375581203.kt)  
[typo_plasma_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.205.3375581203.ts)  
[typo_ruler_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.205.3375581203.kt)  
[typo_ruler_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.205.3375581203.ts)  
[typo_sage_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.205.3375581203.kt)  
[typo_sage_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.205.3375581203.ts)  
[typo_sbermarket_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.205.3375581203.kt)  
[typo_sbermarket_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.205.3375581203.ts)  
[typo_soulmate_kotlin--canary.205.3375581203.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.205.3375581203.kt)  
[typo_soulmate_react-native--canary.205.3375581203.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.205.3375581203.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.110.0-canary.205.3375581203.0
  # or 
  yarn add @salutejs/plasma-b2c@1.110.0-canary.205.3375581203.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

File: packages/plasma-b2c/src/components/Slider/Handle.tsx [hunk 1]
```diff
@@ -44,6 +45,11 @@ const HandleStyledInner = styled.div`
     &:active {
         transform: scale(0.92);
     }
+
+    &:active,
+    &:hover {
+        border-color: ${() => accent};
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/Handle.tsx:51]: А тут точно нужна функция ?

**Comment by Yeti-or** [comment]
Тесты нужно снять еще

File: packages/plasma-b2c/src/components/Slider/Handle.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Handle.tsx:51]: Хорошее замечание! Можно и без нее

**Comment by Yakutoc** [comment]
> Тесты нужно снять еще

А snapshots никак не менялись. Добавлять нечего.

**Comment by Yeti-or** [comment]
Я не понимаю как это, код меняется а тесты нет?
Либо нет такого теста и надо его добавить либо что то идет не так

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Yakutoc** [comment]
> Я не понимаю как это, код меняется а тесты нет? 
> Либо нет такого теста и надо его добавить либо что то идет не так

Тут скорее всего первый вариант. Я понял - добавлю.

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

File: packages/plasma-b2c/src/components/Slider/Handle.tsx [hunk 2]
```diff
@@ -118,7 +130,7 @@ export const Handle = React.forwardRef<HTMLDivElement, HandleProps>(
                 disabled={disabled}
             >
                 <HandleStyled ref={ref} style={{ zIndex }}>
-                    <HandleStyledInner />
+                    <SliderThumb tabIndex={0} data-cy="slider-thumb" />
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Handle.tsx:133]: мы стараемся в библиотеке data-cy аттрибуты не оставлять

File: packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx [hunk 3]
```diff
@@ -21,6 +21,18 @@ describe('plasma-b2c: Slider', () => {
...
     });
 
+    it('default: focus', () => {
+        mount(
+            <CypressTestDecorator>
+                <Slider value={24} min={0} max={100} />
+            </CypressTestDecorator>,
+        );
+
+        cy.get('[data-cy="slider-thumb"]').focus();
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx:31]: тут надо через селекторы

File: packages/plasma-b2c/src/components/Slider/Handle.tsx (hunk 2 above)
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/Handle.tsx:133]: Если компонент `disabled` то фокус на него же нельзя поставить?

File: packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx:31]: > мы стараемся в библиотеке data-cy аттрибуты не оставлять
> тут надо через селекторы

@neretin-trike Но то же хрупкий тест поменялось что-то в разметке - тест упадет.

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

File: packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/Slider/Slider.component-test.tsx:31]: это компромисс между лишней разметкой и поддержкой тестов

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

**Review by Yeti-or — COMMENTED** [review]
Я тут задумался а сточки зрения доступности как должно быть

File: packages/plasma-b2c/src/components/Slider/Handle.tsx [hunk 4]
```diff
@@ -44,6 +47,17 @@ const HandleStyledInner = styled.div`
     &:active {
         transform: scale(0.92);
     }
+
+    &:focus {
+        outline: none;
+    }
+
+    &:not([disabled]) {
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Handle.tsx:55]: Ты прав, нужно там -1 сделать в таком случае. Поправлю - спасибо за внимательность

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-205/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-205/
  ui storybook: http://plasma.sberdevices.ru/pr-205/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-205/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-205/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-205/temple-storybook/

**Pull request merged**